src/tiled/automappingmanager.cpp

```diff
@@ -205,21 +206,53 @@ bool AutomappingManager::loadFile(const QString &filePath)
 void AutomappingManager::setMapDocument(MapDocument *mapDocument)
 {
     cleanUp();
-    if (mMapDocument)
+
+    QString oldRules;
+    if (mMapDocument) {
         mMapDocument->disconnect(this);
+        oldRules = getRulesFileName();
+        QFileInfo oldFile(oldRules);
+        if (oldFile.exists()) {
+            mWatcher->disconnect(SIGNAL(fileChanged(const QString &)));
+            mWatcher->removePath(oldRules);
+        }
```

Comment:
When we move the connect to the constructor, we can unconditionally call `removePath`:
```c++
mWatcher->removePath(rulesFileName());
```
Since it does not matter whether the file exists or not.

We can also do the `removePath`/`addPath` only when the rules file changes.